pointInPoly.ts

```diff
@@ -0,0 +1,185 @@
+      if (xi > xp) {
+        nleft += 1;
+      }
+    }
+  }
+  let xin = nleft % 2;
+  return xin == 1;
+}
+
+// Test all polygons (polys) if a point lies in any
+//    polygons are assumed to be in GeoJSON format and have:
+//      - a bounding box (xmin,ymin,zmin,xmax,ymax,zmax)
+//      - a priority where overlapping polygons with higher priority are chosen
+//    Returns the found polygon or null in the case of no match
+export function findPolygon(p: any, polys: any) {
+  let found = [];
+  for (let j = 0; j < polys.features.length; j++) {
+    const poly = polys.features[j];
+    //console.log(j, poly.properties.group);
+    //if ((poly.properties.ymin && p[1] < poly.properties.ymin) ||
+    //  (poly.properties.ymax && p[1] > poly.properties.ymax)) {
+    //  continue;
+    //}
+    //if (found && poly.properties.priority < found.properties.priority) {
+    //  continue;
+    //}
+    if (isPointInsidePolygon(poly, p)) {
+      //found = polys.features[j];
+      //found.push(poly.properties.group);
+      found.push(j);
+    }
+  }
+  return found;
+}
+
+function vlen2(p: any) {
+  return p[0] * p[0] + p[1] * p[1]
+}
+function vlen(p: any) {
+  return Math.sqrt(vlen2(p))
+}
+
+function pointLineDist(p: any, p1: any, p2: any) {
+  const x1 = p1[0];
+  const y1 = p1[1];
+  const x2 = p2[0];
+  const y2 = p2[1];
+  const x0 = p[0];
+  const y0 = p[2];
+  const dx = x2 - x1;
+  const dy = y2 - y1;
+  const numer = Math.abs(dx * (y1 - y0) - (x1 - x0) * dy);
+  const denom = Math.sqrt(dx * dx + dy * dy);
+  return numer / denom;
+}
+
+function sqr(x: any) { return x * x }
+function dist2(v: any, w: any) { return sqr(v.x - w.x) + sqr(v.y - w.y) }
+function distToSegmentSquared(p: any, v: any, w: any) {
+  var l2 = dist2(v, w);
+  if (l2 == 0) return dist2(p, v);
+  var t = ((p.x - v.x) * (w.x - v.x) + (p.y - v.y) * (w.y - v.y)) / l2;
+  t = Math.max(0, Math.min(1, t));
+  return dist2(p, {
+    x: v.x + t * (w.x - v.x),
+    y: v.y + t * (w.y - v.y)
+  });
+}
+function distToSegment(p: any, v: any, w: any) {
+  return Math.sqrt(distToSegmentSquared({ x: p[0], y: p[2] },
+    { x: v[0], y: v[1] },
+    { x: w[0], y: w[1] }));
+}
+
+function distToPolygon(p: any, poly: any) {
+  let minDist = 1e7;
+  const n = poly.geometry.coordinates[0].length;
+  for (let i = 0; i < n; i++) {
+    const p1 = poly.geometry.coordinates[0][i];
+    const p2 = poly.geometry.coordinates[0][(i + 1) % n];
+    const dist = distToSegment(p, p1, p2);
+    if (dist < minDist) {
+      minDist = dist;
+    }
+  }
+  return minDist;
+
+}
+
+export function findPolygons(p: any, polys: any, minDist: number) {
+  let out = findPolygon(p, polys);
+  if (out.length > 0) {
+    return out;
+  }
+  for (let j = 0; j < polys.features.length; j++) {
+    const poly = polys.features[j];
+    const dist = distToPolygon(p, poly);
+    if (dist < minDist) {
+      out.push(j)
+    }
+  }
+  return out;
+}
+
+export function findClosestPolygon(p: any, polys: any, minDist: number) {
```

Comment:
it looks like this returns an array of polygons instead of just a polygon; shouldn't this be findClosestPolygon**s** then?

Reply:
Probably, but I am going to remove it :)